chain/chain/src/chain.rs

```diff
@@ -5099,7 +5102,17 @@ impl<'a> ChainUpdate<'a> {
 pub fn do_apply_chunks(
     work: Vec<Box<dyn FnOnce() -> Result<ApplyChunkResult, Error> + Send>>,
 ) -> Vec<Result<ApplyChunkResult, Error>> {
-    work.into_par_iter().map(|task| task()).collect::<Vec<_>>()
+    let parent_span = tracing::debug_span!(target: "chain", "Apply chunks").entered();
+    work.into_par_iter()
+        .map(|task| {
+            // As chunks can be processed in parallel, make sure they are all tracked as children of
+            // a single span.
+            let _span =
+                tracing::debug_span!(target: "chain", parent: &parent_span, "Chunk").entered();
```

Comment:
I wonder if you could also add the shard_id here ?

Reply:
Made a bit of a refactoring to parse `parent_span` to the `task()`. Let me know if this makes sense.